lsp/server/src/server.ts

```diff
@@ -85,65 +101,40 @@ connection.onInitialized(() => {
     }
 });
 
-// Settings for Mobile LSP
-export interface MobileSettings {
-    maxNumberOfProblems: number; //max number of diagnostics to detect per document.
-}
-
-// The global settings, used when the `workspace/configuration` request is not supported by the client.
-// Please note that this is not the case when using this server with the client provided
-// but could happen with other clients.
-const defaultSettings: MobileSettings = { maxNumberOfProblems: 1000 };
-let globalSettings: MobileSettings = defaultSettings;
+connection.onDidChangeConfiguration((change) => {
+    const changedSetting =
```

Comment:
when settings under salesforce.salesforcedx-vscode-mobile.diagnostics are changed, get the new settings.